ispcrt/detail/gpu/GPUDevice.cpp

```diff
@@ -1835,7 +1859,7 @@ GPUDevice::GPUDevice() : GPUDevice(nullptr, nullptr, 0) {}
 
 GPUDevice::GPUDevice(void *nativeContext, void *nativeDevice, uint32_t deviceIdx) {
     // Perform GPU discovery
-    m_driver = gpu::deviceDiscovery(&m_is_mock);
```

Comment:
When a `nativeDevice` is passed then `m_driver` keeps being `nullptr`, which is problematic for callers of `platformNativeHandle()`. Can in this case the `gpu::g_deviceList` be searched for the `nativeDevice` to set `m_driver` as well?